### ISSUE TITLE
Can't define properties that does not match the index type

### ISSUE THREAD

**Issue by Pajn** [Microsoft/TypeScript#7323]
I'm trying to define the type
```typescript
type Node = {
    $name: string
    $text: string
    [childTagName: string]: Node
}
```

**Actual behavior:** 
The error "Property '$name' is not assignable to index type '{ [ childTagName: string ]: Node; $name: string; $text: string }' is reported by the compiler.

**Expected behavior:**
I should be able to define a type with some defined properties of one type and then any number of properties of a different type. Now it seams like Typescript does not allow these "mixed" types.

**TypeScript Version:** 
1.8.2

**Comment by sandersn** [comment]
This is by design. The string indexer says that any index will produce something of type Node:

```ts
let n: Node;
n['foo'] // must be of type node
n['$name'] // must also be of type node
```

That means any extra properties in a type with a string indexer must be the same as the type of the indexer.
